### PR TITLE
Bookstore reducers and actions

### PR DESCRIPTION
**Pull request by Bettenoch**
<h2 dir="auto">Made the following changes:♻</h2>

🔹Install react-redux @reduxjs/toolkit.
🔸Create a directory that will contain all your Redux logic
🔹Configure a Redux store.
🔸Create a slice of state for books that includes an array of books (initial state: empty array), Includes a reducer that adds a book, and Includes a reducer that removes a book.
🔹Create  a slice of state for categories that includes an array of categories (initial state: empty array) and Includes a reducer that checks the status.

<hr></hr>
<h3 dir="auto">Thank your for reviewing my pull request👍.</h3>

### CODE REVIEW

File: src/Redux/books/bookSlice.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    case DELETE_BOOK:
+      return [
+        ...state.filter((item) => item.id !== action.id),
+      ];
+    default:
+      return state;
+  }
+}
+
+export const addBook = (book) => ({ type: ADD_BOOK, payload: book });
```
Comment (ghost) [inline src/Redux/books/bookSlice.js]: - [ ] Fantastic job on your state management implementation using individual action types and action creators! However, for improved code simplicity and better alignment with project guidelines, I'd suggest adopting Redux Toolkit's createSlice approach. It offers automatic action creator generation, concise syntax, and enhances maintainability. Here is an example implementation.

```
// ./src/redux/books/booksSlice.js

import { createSlice } from "@reduxjs/toolkit"

export const booksSlice = createSlice({
  name: "books",
  initialState: {
    books: [],
  },
  reducers: {
    addBook: (state, action) => {
      state.books = [
        ...state.books,
        {
          id: Math.floor(Math.random() * 100),
          title: action.payload.title,
          author: action.payload.author,
        },
      ]
    },
    removeBook: (state, action) => {
      state.books = state.books.filter((book) => book.id !== action.id)
    },
  },
})

export const { addBook, removeBook } = booksSlice.actions

export default booksSlice.reducer


```

File: src/Redux/categories/categoriesSlice.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+    case CHECK_STATUS:
+      return 'Under Construction';
+
+    default:
+      return state;
+  }
+}
+
+export const checkCurStatus = () => ({
+  type: CHECK_STATUS,
```
Comment (ghost) [inline src/Redux/categories/categoriesSlice.js]: - [ ] The same case also applies to your categoriesSlice. Here is an example code.

```
// ./src/redux/categories/categoriesSlice.js

import { createSlice } from "@reduxjs/toolkit"

export const categoriesSlice = createSlice({
  name: "categories",
  initialState: {
    categories: [],
  },
  reducers: {
    checkStatus: (state, action) => {
      state.categories =
        action.payload === "Under construction"
          ? "Under construction"
          : state.categories
    },
  },
})

export const { checkStatus } = categoriesSlice.actions

export default categoriesSlice.reducer

```

 - [ ] Keep up the excellent work! 🚀

**Review by ghost — CHANGES_REQUESTED** [review]
Hi @Bettenoch , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!




## Highlights

✅ Correct use of Gitflow
✅ No linter error
✅ Professional documentation 👍 
✅ The structure of the files follows `ducks pattern`: 👍 


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

N/A
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better.
Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Davidosky007 — APPROVED** [review]
Hi @Bettenoch 👋, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
* Redux store setup properly 👍 
* No linter errors 👍 
* Well written PR 👍 
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**